src/openms/source/FILTERING/NOISEESTIMATION/SignalToNoiseEstimator.cpp

```diff
@@ -29,7 +29,47 @@
 //
 // --------------------------------------------------------------------------
 // $Maintainer: Chris Bielow $
-// $Authors: $
+// $Authors: Chris Bielow $
 // --------------------------------------------------------------------------
 //
 
+#include <OpenMS/FILTERING/NOISEESTIMATION/SignalToNoiseEstimator.h>
+
+#include <OpenMS/KERNEL/MSExperiment.h>
+
+#include <random>
+
+using namespace std;
+
+namespace OpenMS
+{
+
+  float estimateNoiseFromRandomMS1Scans(const MSExperiment& exp, const UInt n_scans, const double percentile)
+  {
+    vector<Size> ms1_indices;
+    for (Size i = 0; i < exp.size(); ++i) if (exp[i].getMSLevel() == 1 && !exp[i].empty()) ms1_indices.push_back(i);
+
+    if (ms1_indices.empty()) return 0.0f;
+
+    std::default_random_engine generator(time(nullptr));
```

Comment:
might not be protable / will introduce non-determinism

Reply:
should be very portable. Just not deterministic